mvvm-components/src/test/kotlin/se/springworks/mvvmcomponents/fragment/BaseDataBindingFragmentTest.kt

```diff
@@ -0,0 +1,70 @@
+package se.springworks.mvvmcomponents.fragment
+
+import android.view.View
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.spy
+import com.nhaarman.mockito_kotlin.verify
+import com.vhudnitsky.mvvmcomponents.databinding.StringItemBinding
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert
+import org.junit.Before
+import org.junit.Test
+import se.springworks.mvvmcomponents.recyclerview.viewmodel.StringItemViewModel
+
+class StubBaseDataBindingFragment : BaseDataBindingFragment<StringItemBinding, StringItemViewModel>() {
+  override fun resourceBRViewModelId(): Int {
+    return 0
+  }
+
+  override fun provideViewModel(): StringItemViewModel {
+    return StringItemViewModel()
+  }
+
+  override fun getLayoutId(): Int {
+    return 0
+  }
+}
+
+class BaseDataBindingFragmentTest {
+  private lateinit var fragment: StubBaseDataBindingFragment
+
+  @Before
+  fun setUp() {
+    fragment = StubBaseDataBindingFragment()
+    fragment.onCreate(null)
+  }
+
+  @Test
+  fun testOnCreateInitViewModel() {
+    MatcherAssert.assertThat(fragment.viewModel, CoreMatchers.notNullValue())
+  }
+
+  @Test
+  fun testOnViewCreatedInitializeViewModel() {
+    fragment.onViewCreated(mock<View>(), null)
+    val viewModelSpy = spy(fragment.viewModel)
```

Comment:
Should this be moved up one line?

Reply:
Is there some sort of compile time magic that sets the viewmodel in the fragment to a spy when `spy(fragment.viewModel)` is called?

Reply:
No no view models created inside a fragment. And it's not a mock, but real one. So I need to verify that life cycle methods call exactly the same methods for view model

Reply:
Yeah, but how does calling `spy(fragment.viewModel)` after `fragment.onViewCreated(...)` has been called capture anything?